python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -561,9 +569,66 @@ def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationRe
         )
 
 
+@whitelist_for_serdes
+@experimental
+class AutoMaterializeAssetPartitionFilter(
+    NamedTuple(
+        "_AutoMaterializeAssetPartitionFilter",
+        [("latest_materialization_run_forbidden_tag_keys", Optional[FrozenSet[str]])],
+    )
+):
+    """A filter that can be applied to an asset partition, during auto-materialize evaluation, and
+    returns a boolean for whether it passes.
+
+    Attributes:
+        latest_materialization_run_forbidden_tag_keys (Optional[Sequence[str]]): `passes` returns
+            False if the run responsible for the latest materialization of the asset partition has
+            any of these tag keys.
+    """
+
+    def __new__(cls, latest_materialization_run_forbidden_tag_keys: Optional[AbstractSet[str]]):
+        return super().__new__(
+            cls,
+            latest_materialization_run_forbidden_tag_keys=frozenset(
+                latest_materialization_run_forbidden_tag_keys
+            )
+            if latest_materialization_run_forbidden_tag_keys
+            else None,
+        )
+
+    def passes(self, context: RuleEvaluationContext, asset_partition: AssetKeyPartitionKey) -> bool:
+        if asset_partition in context.will_materialize_mapping.get(
+            asset_partition.asset_key, set()
+        ):
+            return False
+
+        materialization_record = (
+            context.instance_queryer.get_latest_materialization_or_observation_record(
+                asset_partition, after_cursor=context.cursor.latest_storage_id
+            )
+        )
+        if materialization_record is None:
+            check.failed(
+                f"AutoMaterializeAssetPartitionFilter was invoked for asset partition {asset_partition},"
+                " which was not updated since previous tick. It should only be invoked for asset"
+                " partitions updated since the previous tick."
+            )
+
+        return all(
+            not context.instance_queryer.run_has_tag(
+                materialization_record.run_id, run_tag_key, None
+            )
+            for run_tag_key in self.latest_materialization_run_forbidden_tag_keys or set()
+        )
+
+
 @whitelist_for_serdes
 class MaterializeOnParentUpdatedRule(
-    AutoMaterializeRule, NamedTuple("_MaterializeOnParentUpdatedRule", [])
+    AutoMaterializeRule,
+    NamedTuple(
+        "_MaterializeOnParentUpdatedRule",
+        [("updated_parent_filter", Optional[AutoMaterializeAssetPartitionFilter])],
```

Comment:
the addition of a new parameter should modify the description property, as users should have some way, via the UI, to understand that this is different from a non-customized rule. Doesn't necessarily need to be super clean, maybe just:

"upstream data has changed since the latest materialization, subject to the filter: 'latest materialization of parent must not have any of the following run tag keys: {...}'."

Pretty ugly, but only shows up if you're in this niche area.